webplugin/js/app/media/media-service.js

```diff
@@ -39,43 +39,48 @@ Kommunicate.mediaService = {
                 // stop mic effect
                 Kommunicate.typingAreaService.hideMiceRecordingAnimation();
                 window.$applozic.fn.applozic('toggleMediaOptions');
-;            }
+                ;
+            }
         }
     },
     voiceOutputIncomingMessage: function (message) {
-        // get appoptions
-        var appOptions = KommunicateUtils.getDataFromKmSession("appOptions");
+        try {
+            // get appoptions
+            var appOptions = KommunicateUtils.getDataFromKmSession("appOptions");
 
-        // If the message isn't part of the UI, it's not included
-        // in voiceoutput either
-        if (!Kommunicate.visibleMessage(message)) return;
+            // If the message isn't part of the UI, it's not included
+            // in voiceoutput either
+            if (!Kommunicate.visibleMessage(message)) return;
 
-        // if voiceoutput is enabled and browser supports it
-        if (appOptions.voiceOutput && "speechSynthesis" in window) {
-            var textToSpeak = "";
-            if (message.hasOwnProperty("fileMeta")) {
-                textToSpeak += MCK_LABELS['voice.output'].attachment;
-                textToSpeak += message.fileMeta.name;
-            }
-            else if (message.contentType == KommunicateConstants.MESSAGE_CONTENT_TYPE.LOCATION) {
-                coord = JSON.parse(message.message);
-                textToSpeak += MCK_LABELS['voice.output'].location.init;
-                textToSpeak += MCK_LABELS['voice.output'].location.lat + Math.round(coord.lat * 100) / 100;
-                textToSpeak += MCK_LABELS['voice.output'].location.lon + Math.round(coord.lon * 100) / 100;
-            }
-            else if (message.message) {
-                textToSpeak += message.message;
-            }
-            if (textToSpeak) {
-                var utterance = new SpeechSynthesisUtterance(textToSpeak);
-                utterance.lang = appOptions.language || "en-US";
-                utterance.onerror = function (event) {
-                    if (event.error !== "not-allowed") {
-                    throw new Error("Error while converting the message to voice.", event.error);
-                    }
-                };
-                speechSynthesis.speak(utterance);
+            // if voiceoutput is enabled and browser supports it
+            if (appOptions.voiceOutput && "speechSynthesis" in window) {
```

Comment:
@shubhamtibra add check for "appOptions".
Debug the issue and check why appOptions is coming null/undefined.
Adding try/catch won't solve the problem here. It will just hide the issue and not break the current flow.

Reply:
Okay, I will add check for `appOptions` as in the previous commit. Should I remove the try...catch or keep it, just in case the code throws some other error which might cause the flow break.

Reply:
I will check what cases error can occur and update according to that.

Reply:
i got this error in jsfiddle, see it in that once